### ISSUE TITLE
I got an error  when  I use  `flipTo` function in a 'react' project

### ISSUE THREAD

**Issue by Jickey** [gaoryrt/number-flip#25]
my react component code:
`const FlipNumber = ({ number: realNumber, className }) => {`
`  const ref = useRef();`
`  const flipRef = useRef(null);`

`  useEffect(() => {`
 `   if (!flipRef.current) {`
`      flipRef.current = new Flip({`
`       node: ref.current,`
`        from: 0,`
`        to: realNumber,`
`        duration: 1.5,`
`        separator: ',',`
`      });`
`    } else {`
`      // error when flipTo runs `
`      flipRef.current.flipTo({ to: realNumber });`
`    }`
`  }, [realNumber]);`

`  return (`
`   <div`
`      ref={ref}`
`    />`
`  );`
`};`

![image](https://user-images.githubusercontent.com/23648316/147060971-52a5b5eb-8703-492d-97e6-c95c67def235.png)

![image](https://user-images.githubusercontent.com/23648316/147061040-914068c4-dfc6-4375-baf6-18a72f814cfc.png)

**Comment by gaoryrt** [comment]
I think this will work:
https://codesandbox.io/s/snowy-tdd-gf9hp?file=/src/App.js:89-488
```jsx
const FlipNumber = ({ number: realNumber }) => {
  const ref = useRef();
  const flipRef = useRef(null);
  useEffect(() => {
    flipRef.current = new Flip({
      node: ref.current,
      from: 0,
      to: realNumber,
      duration: 1.5,
      separator: ","
    });
  }, []);
  useEffect(() => {
    flipRef.current.flipTo({ to: realNumber });
  }, [realNumber]);

  return <div ref={ref} />;
};
```

**Comment by Jickey** [comment]
Yeah. It does work in my  'vite' demo project but I failed in my production project, which meas it may be an incompatibility problem.  Forget it.  Thanks very much for your reply  !

**Comment by xmsz** [comment]
any updated?

**Comment by gaoryrt** [comment]
@xmsz I believe [this](#issuecomment-999991247) works effectively

**Comment by xmsz** [comment]
> @xmsz I believe [this](#issuecomment-999991247) works effectively

<img width="1007" alt="image" src="https://user-images.githubusercontent.com/5879146/206084095-9dc7cabd-7ae6-441d-aa01-dc6894c7da5e.png">

**Comment by edgar-505** [comment]
@gaoryrt still error...

**Comment by gaoryrt** [comment]
> @gaoryrt still error...
Please provide a minimal reproduction.

**Comment by TanYiBing** [comment]
似乎是数字位数不一样，所以提示 Cannot read properties of undefined (reading 'childNodes')

**Comment by eviltik** [comment]
On my side, fixed by adding "?"

```
for (let i = 0; i < this.ctnrArr[digit]?.childNodes.length; i += 1) {
```

Sorry, no time to understand exactly the problem and no time for PR.

I'm using number-flip with lit-element (shadowdom)

Btw thank you for this cool lib ;)

**Comment by eviltik** [comment]
Mhhh bad fixe. The problem occured only when i'm using a 2 digit number (from:0, to: 10)

**Comment by eviltik** [comment]
More related to #27

**Comment by gaoryrt** [comment]
@eviltik 
Yes, due to the design, each digit in the flipping number must remain it's unique place.
Like if you start with a 3-digit number, say 742, the biggest number you can go in the later flipping is 999.
